File: com.strategypattern/python/SortingContext.py
from SortingStrategy import SortingStrategy

class SortingContext(object):
    
    Strategy: SortingStrategy
    
    def __init__(self, strategy):
        self.Strategy = strategy

    def set_sorting_strategy(self, strategy):
        self.Strategy = strategy
        
    def perform_sort(self, data) -> list:
        return self.Strategy.sort(data)

File: Strategy/ts/SortingContext.ts
import { SortingStrategy } from "./SortingStrategy";

export class SortingContext {
    private strategy: SortingStrategy;

    constructor(strategy: SortingStrategy) {
        this.strategy = strategy;
    }

    public setStrategy(strategy: SortingStrategy): void {
        this.strategy = strategy;
    }

    public sort(data: number[]): void {
        this.strategy.sort(data);
    }
}